### ISSUE TITLE
CurrencyTextBox does not select value when tabbed to in Chrome

### ISSUE THREAD

**Issue by schallm** [dojo/dijit#158]
CurrencyTextBox does not keep selection like other text boxes do when tabbed to if `currency` property is specified in dojo 1.10.8 or current (1.13.0).  It seems to be a race condition when reformatting without the currency symbol.  Works as expected in current versions Edge and Firefox.

[Example](https://next.plnkr.co/edit/BqyajhmX3UsE94BRolce?preview)

1. Click to set focus in one of the text boxes below.
2. Use Tab and Shift+Tab to navigate using the keyboard.
3. Notice that the CurrencyTextBox does not keep the contents selected when tabbed to if `currency` property is specified

```html
    <div class="control-group">
        <label for="CurrencyTextBox">Currency Text Box:</label>
        <input data-dojo-type="dijit/form/CurrencyTextBox" id="CurrencyTextBox" />
    </div>
    <div class="control-group">
        <label for="EURCurrencyTextBox">EUR Currency Text Box:</label>
        <input data-dojo-type="dijit/form/CurrencyTextBox" id="EURCurrencyTextBox" data-dojo-props="currency: 'EUR'" />
    </div>
    <div class="control-group">
        <label for="USDCurrencyTextBox">USD Currency Text Box:</label>
        <input data-dojo-type="dijit/form/CurrencyTextBox" id="USDCurrencyTextBox" data-dojo-props="currency:'USD'" />
    </div>
```

**Comment by schallm** [comment]
The issue seems to be happen in the following code in `focus.js` when the `CurrencyTextBox` has to update the textbox's value by removing the currency symbol in `_onFocus`.  Again, in my testing, only Chrome has issues with this functionality.

```js
	// for all element that have come into focus, set focused=true
	for(i++; i <= lastNewIdx; i++){
		widget = registry.byId(newStack[i]);
		if(widget){
			widget.set("focused", true);
			if(widget._focusManager == this){
				widget._onFocus(by);
			}
			this.emit("widget-focus", widget, by);
		}
	}
```

I found a workaround for by listening for the `widget-focus` and calling textbox.select if it is the correct widget type.

```js
	workaroundHandle = focusUtil.on("widget-focus", function(widget, by){
		if (by === undefined && widget.isInstanceOf(CurrencyTextBox)) {
			widget.textbox.select();
		}
	});
```

I updated the [example](https://next.plnkr.co/edit/BqyajhmX3UsE94BRolce?preview) and you can toggle the workaround.

Horrible solution, but it works.   Any other ideas?

**Comment by wkeese** [comment]
I don't know, it's been so long since I even looked at this code.  I'm assuming this happens with NumberTextBox too because they share the same... so my guess would be to just update NumberTextBox#_onFocus() to call select() after setting the value.

**Comment by wkeese** [comment]
Ah good point about the mouse click.   I don't think it matters much if you use `has("chrome")` or do it for all browsers.  Doing the latter will avoid the wrath of people fanatically against browser sniffing.

**Comment by dylans** [comment]
Fixed in c608818. Thanks!
